src/linker/Linker.Dataflow/MethodBodyScanner.cs

```diff
@@ -0,0 +1,825 @@
+				// old value.
+				newValue.Value = MergePointValue.MergeValues (existingValue.Value, valueToStore);
+			}
+			valueCollection [collectionKey] = newValue;
+		}
+
+		public void Scan (MethodBody methodBody)
+		{
+			MethodDefinition thisMethod = methodBody.Method;
+
+			Dictionary<VariableDefinition, ValueBasicBlockPair> locals = new Dictionary<VariableDefinition, ValueBasicBlockPair> (methodBody.Variables.Count);
+
+			Dictionary<int, Stack<StackSlot>> knownStacks = new Dictionary<int, Stack<StackSlot>> ();
+			Stack<StackSlot> currentStack = new Stack<StackSlot> (methodBody.MaxStackSize);
+
+			ScanExceptionInformation (knownStacks, methodBody);
+
+			BasicBlockIterator blockIterator = new BasicBlockIterator (methodBody);
+
+			MethodReturnValue = null;
+			foreach (Instruction operation in methodBody.Instructions) {
+				int curBasicBlock = blockIterator.MoveNext (operation);
+
+				if (knownStacks.ContainsKey (operation.Offset)) {
+					if (currentStack == null) {
+						// The stack copy constructor reverses the stack
+						currentStack = new Stack<StackSlot> (knownStacks [operation.Offset].Reverse ());
+					} else {
+						currentStack = MergeStack (currentStack, knownStacks [operation.Offset], methodBody, operation.Offset);
+					}
+				}
+
+				if (currentStack == null) {
+					currentStack = new Stack<StackSlot> (methodBody.MaxStackSize);
+				}
+
+				switch (operation.OpCode.Code) {
+					case Code.Add:
+					case Code.Add_Ovf:
+					case Code.Add_Ovf_Un:
+					case Code.And:
+					case Code.Div:
+					case Code.Div_Un:
+					case Code.Mul:
+					case Code.Mul_Ovf:
+					case Code.Mul_Ovf_Un:
+					case Code.Or:
+					case Code.Rem:
+					case Code.Rem_Un:
+					case Code.Sub:
+					case Code.Sub_Ovf:
+					case Code.Sub_Ovf_Un:
+					case Code.Xor:
+					case Code.Cgt:
+					case Code.Cgt_Un:
+					case Code.Clt:
+					case Code.Clt_Un:
+					case Code.Ldelem_I:
+					case Code.Ldelem_I1:
+					case Code.Ldelem_I2:
+					case Code.Ldelem_I4:
+					case Code.Ldelem_I8:
+					case Code.Ldelem_R4:
+					case Code.Ldelem_R8:
+					case Code.Ldelem_U1:
+					case Code.Ldelem_U2:
+					case Code.Ldelem_U4:
+					case Code.Shl:
+					case Code.Shr:
+					case Code.Shr_Un:
+					case Code.Ldelem_Any:
+					case Code.Ldelem_Ref:
+					case Code.Ldelema:
+					case Code.Ceq:
+						PopUnknown (currentStack, 2, methodBody, operation.Offset);
+						PushUnknown (currentStack, 1);
+						break;
+
+					case Code.Dup:
+						currentStack.Push (currentStack.Peek ());
+						break;
+
+					case Code.Ldnull:
+						currentStack.Push (new StackSlot (NullValue.Instance));
+						break;
+
+					
+					case Code.Ldc_I4_0:
+					case Code.Ldc_I4_1:
+					case Code.Ldc_I4_2:
+					case Code.Ldc_I4_3:
+					case Code.Ldc_I4_4:
+					case Code.Ldc_I4_5:
+					case Code.Ldc_I4_6:
+					case Code.Ldc_I4_7:
+					case Code.Ldc_I4_8: {
+							int value = operation.OpCode.Code - Code.Ldc_I4_0;
```

Comment:
I am not sure I like this. Cecil can change the values anytime

Reply:
I don't think that's likely. The enum is directly mirroring the values in the CIL specification.

Reply:
> Cecil can change the values anytime

Changing the value of an enum member is a breaking source and binary change.

I could see Cecil being forced into a breaking change if we ever add new unprefixed IL instructions (Cecil currently assumes all IL instructions fit in a single byte, and folds the ones that have a twobyte encoding into a single byte crossing fingers there's enough space left). But I wouldn't expect these to be ever _reordered_ - complex lookup tables from physical opcode encoding to `Code` would introduce unnecessary inefficiencies when parsing.

Reply:
Cecil is not API stable so even source breaking changes can happen (so just be careful)